java-squid/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -127,4 +129,16 @@ public String toString() {
     return "{" + values.toString() + "}  {" + constraints.toString() + "}" + " { " + stack.toString() + " }";
   }
 
+  public boolean isReachable(final SymbolicValue sv) {
+    class ContainsValue implements PMap.Consumer<Symbol, SymbolicValue> {
+      boolean result = false;
+      @Override
+      public void accept(Symbol key, SymbolicValue value) {
+        result |= sv == value;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)